src/snowflake/snowpark/dataframe.py

```diff
@@ -1124,16 +1132,15 @@ def select_expr(self, *exprs: Union[str, Iterable[str]]) -> "DataFrame":
             <BLANKLINE>
 
         """
-        return self.select(
-            [sql_expr(expr) for expr in parse_positional_args_to_list(*exprs)]
-        )
+        parsed_exprs: List[str] = parse_positional_args_to_list(*exprs)
+        return self.select([sql_expr(expr) for expr in parsed_exprs])
 
     selectExpr = select_expr
 
     @df_api_usage
     def drop(
         self,
-        *cols: Union[ColumnOrName, Iterable[ColumnOrName]],
+        *cols: ColumnOrName,
```

Comment:
just curious about why we remove the Iterable for the *args parameter, is this mypy standard?